nemo/collections/nlp/models/language_modeling/megatron_gpt_model.py

```diff
@@ -534,6 +539,33 @@ def initialize_ub_func(self):
         )
         self.initialize_ub = False
 
+    def on_train_batch_start(self, batch, batch_idx):
+        super().on_train_batch_start(batch, batch_idx)
+        P = parallel_state.get_pipeline_model_parallel_rank()
+        T = parallel_state.get_tensor_model_parallel_rank()
+        D = parallel_state.get_data_parallel_rank()
+        R = torch.distributed.get_rank()
```

Comment:
What is the purpose of creating these constants?

Reply:
Replaced them with more clearly named local variables. They are used in the NVTX range annotation below.